cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/sql/PostgreSqlStructuredTableAdmin.java

```diff
@@ -186,6 +229,37 @@ private List<String> getCreateIndexStatements(StructuredTableId tableId, Set<Str
     return statements;
   }
 
+  private Optional<String> getAddColumnStatement(StructuredTableId tableId,
+                                                 DatabaseMetaData metaData,
+                                                 List<FieldType> nonPrimaryKeyFields)
+    throws SQLException {
+    List<String> existingFields = new ArrayList<>();
+    ResultSet resultSet = metaData.getColumns(null, null, tableId.getName(), null);
+    while (resultSet.next()) {
+      existingFields.add(resultSet.getString("COLUMN_NAME"));
+    }
+
+    List<FieldType> fieldsToAdd = nonPrimaryKeyFields
+      .stream()
+      .filter(nonPrimaryKeyField -> !existingFields.contains(nonPrimaryKeyField.getName()))
+      .collect(Collectors.toList());
+
+    if (fieldsToAdd.size() == 0) {
```

Comment:
Use `isEmpty()` instead of `size() == 0`

Reply:
updated